### DISCUSSION TITLE
add async tls handshake

### DISCUSSION THREAD

**Discussion by calvin2021y** [curl/curl#12594] [Q&A]
I use curl multi handle with libuv event loop.    like (https://curl.se/libcurl/c/multi-uv.html)

Recent I debug a windows problem(https://github.com/curl/curl/issues/12583), notify the loop is blocked frequently.

I am not sure this is a new or old problem.   the best way to implement this maybe add some libuv async task to delegate the blocked tls handshake  task into worker thread.

I notice the block time is like over 1s,  I am not sure tls handshake can take so much time.

I also suspect some curl logic cause the block, but not find a way to prove it yet.

**Comment by bagder** [comment]
> I notice the block time is like over 1s, I am not sure tls handshake can take so much time.

If it indeed does that, it is a bug.
